src/lib/selections.ts

```diff
@@ -101,38 +109,74 @@ export async function getEventForSelectWithPoint(
       resp?.resp?.data?.modeling_response?.type === 'entity_get_parent_id'
         ? resp?.resp?.data?.modeling_response?.data?.entity_id
         : ''
-    const parentArtifact = engineCommandManager.artifactMap[parentId]
+    const parentArtifact = engineCommandManager.artifactGraph.get(parentId)
     if (parentArtifact) {
       _artifact = parentArtifact
     }
   }
-  const sourceRange = _artifact?.range
   if (_artifact) {
```

Comment:
Nit: We could probably remove a level of indentation using ? on the accesses below and do a if(!_artifact) return null at the very top

Reply:
Good call, I just realised that the "no artifact so ask the engine about it's parent", was really only there for the solid2d that's not in the graph, so I clean up more as well.

https://github.com/KittyCAD/modeling-app/pull/3140/commits/3c3caf84697961ef2b7e0f53d4221dc8f6c63ac6